### PR TITLE
Implement Electron security guidelines

### PR DESCRIPTION
**Pull request by amaurymartiny**
Electron has a 12-point checklist called "[Security, Native Capabilities, and Your Responsibility](https://electronjs.org/docs/tutorial/security)". Parity UI should follow those guidelines.

- [x] **Only load secure content**
- [x] **Disable the Node.js integration in all renderers that display remote content**
- [x] **Enable context isolation in all renderers that display remote content**. 
- [x] **Use ses.setPermissionRequestHandler() in all sessions that load remote content**
- [x] **Do not disable webSecurity**
- [x] **Define a Content-Security-Policy and use restrictive rules (i.e. script-src 'self')**.
- [ ] **Override and disable eval, which allows strings to be executed as code.** Not enabling this one as Web3 Console dapp needs `eval()`. However all networks dapps disallow eval() in their CSP, so we actually don't need this.
- [x] **Do not set allowRunningInsecureContent to true**
- [x] **Do not enable experimental features**
- [x] **Do not use blinkFeatures**
- [x] **WebViews: Do not use allowpopups**
- [x] **WebViews: Verify the options and params of all <webview> tags**

Also removing the proxy to `/parity-utils`, so that the only `inject.js` the dapps are receiving now is the one injected by Electron.

### CODE REVIEW

**Comment by amaurymartiny** [comment]
"**Enable context isolation in all renderers that display remote content**" needs some explanation (for future reference).

**TL;DR** There are some gymnastics to do in order to have Parity UI follow all these guidelines.

Without `contextIsolation` (i.e. before this PR), I did:
```javascript
// In inject.js
window.ethereum = ...;

// In dapp
console.log(window.ethereum); // Will print something
```

With `contextIsolation`, we now have:
```javascript
// In inject.js
window.ethereum = ...;

// In dapp
console.log(window.ethereum); // undefined, because preload and dapp do not share context
// Also, dapp cannot send IPC message to shell anymore (due to no nodeintegration)
```
1. How to inject window.ethereum in the dapp?
2. How to make dapp and shell communicate? IPC and postMessage don't work between those two.
---

### Solution

1. Easy, a preload script can call `webFrame.executeJavaScript(...)`. So instead of injecting directly `inject.js` (which wouldn't work because of above), we inject `preload.js`, which will do `webFrame.executeJavaScript(/* the content of inject.js as a huge string*/)`. Hence the introduction of preload.js.

2. After asking [here](https://github.com/electron/electron/issues/13130) the solution seems to be having preload.js as a proxy:

`Electron` <---`IPC`---> `preload.js` <---`postMessage`---> `dapp`

Because the shell can't communicate with the dapp via postMessage nor IPC. But the shell can communicate with preload.js via IPC, and the dapp can communicate with preload.js via postMessage. So preload.js acts as a message relayer.

File: electron/index.js [hunk 1]
```diff
@@ -74,6 +74,41 @@ function createWindow () {
     callback({ requestHeaders: details.requestHeaders });
   });
 
+  // Do not accept all kind of web permissions (camera, location...)
+  // https://electronjs.org/docs/tutorial/security#4-handle-session-permission-requests-from-remote-content
+  session.defaultSession
+    .setPermissionRequestHandler((webContents, permission, callback) => {
+      if (!url.startsWith('file:')) {
```
Comment (axelchalon) [inline electron/index.js]: `url` refers to the library here. You forgot: `const url = webContents.getURL()`

Comment (amaurymartiny) [reply electron/index.js]: F*ck... Nice catch, thanks!

File: src/preload.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+// postMessage.
+ipcRenderer.on('PARITY_SHELL_IPC_CHANNEL', (_, data) => window.postMessage(data, '*'));
+// If we receive a postMessage from the webview, we transfer it to the renderer
+// as an IPC message.
+window.addEventListener('message', (event) => {
+  ipcRenderer.sendToHost('PARITY_SHELL_IPC_CHANNEL', { data: event.data });
+});
+
+// Load inject.js as a string, and inject it into the webview with executeJavaScript
+fs.readFile(path.join(remote.getGlobal('dirName'), '..', '.build', 'inject.js'), 'utf8', (err, injectFile) => {
```
Comment (axelchalon) [inline src/preload.js]: Could we use `getBuildPath()` from `src/util/host.js` here?

Comment (amaurymartiny) [reply src/preload.js]: Unfortunately not, I was getting a `window.require` is not a function.

Reason is: in preload.js, `isElectron()` is true, however it's not a renderer process so you would do `require('electron')` instead of `window.require('electron')`
